src/main/java/ch/njol/skript/expressions/ExprNumbers.java

```diff
@@ -81,10 +81,18 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			s = f;
 			f = temp;
 		}
-		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : Math.floor(f.doubleValue() - s.doubleValue() + 1);
+		
 		final List<Number> list = new ArrayList<>();
+		
+		
 		final double low = integer ? Math.ceil(s.doubleValue()) : s.doubleValue();
-		for (int i = 0; i < amount; i++) {
+		double decimalCheck = s.doubleValue() % 1;
+		
+		final int decimals = decimalCheck == 0 ? 0 : Double.toString(decimalCheck).length() - 2;
+		
+		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : f.doubleValue() - s.doubleValue() + (10 ^ (decimals * -1));

```

Comment:
Just realised, what are you trying to achieve with ``10 ^ (decimals * -1)``? Are you trying to do ![image](https://user-images.githubusercontent.com/19924575/67857102-e5a10000-fb50-11e9-8286-2bbcad9b5ff2.png) or...?

Reply:
If the decimals are 1, (so for example the start is 2.**_1_**, it'll add 0.1 each time, because 10 ^ (1 * -1) = 10 ^ -1 = 0.1, which is the increment we need.
When there are no decimals (so decimals = 0), we have 10 ^ 0 = 1, which is also the increment we need if the start value is an integer (even when they stated to loop numbers).

There may be a faster way to do this.

Reply:
Not nitpicking at efficiency here, ``10 ^ (decimals * -1)`` is mathematically correct but wrong in syntax - in Java ^ represents the bitwise XOR function, if you want the exponent function, use ``Math.pow(a, b)``

Reply:
But while you're at it, you could clean up the code slightly by caching the value of ``Math.pow(10, decimals * -1)``, to prevent constant re-computation in the declaration of ``amount``, and in the for-loop incrementer